src/feeder/JSONNode.cpp

```diff
@@ -0,0 +1,194 @@
+#include "JSONNode.h"
+
+// JSONNode default constructor
+JSONNode::JSONNode() {}
+
+// JSONNode copy constructor
+JSONNode::JSONNode(const JSONNode& t) {
+  node_id = t.node_id;
+  node_name = t.node_name;
+  node_type = t.node_type;
+  is_cpu_op = t.is_cpu_op;
+  runtime = t.runtime;
+  data_deps = t.data_deps;
+  dep_unresolved_parent_ids_json = t.dep_unresolved_parent_ids_json;
+  children_vec_json = t.children_vec_json;
+  children_set_json = t.children_set_json;
+  /* 	COMM_SEND_NODE = 5,
+          COMM_RECV_NODE = 6,
+          COMM_COLL_NODE = 7 	*/
+  if (node_type == 5 || node_type == 6 || node_type == 7) {
+    tensor_size = t.tensor_size;
+    comm_type = t.comm_type;
+    comm_priority = t.comm_priority;
+    comm_size = t.comm_size;
+    comm_src = t.comm_src;
+    comm_dst = t.comm_dst;
+    comm_tag = t.comm_tag;
+  }
+}
+
+// JSONNode constructor
+JSONNode::JSONNode(json data, uint64_t id) {
+  try {
+    node_id = data["workload_graph"][id]["Id"];
+  } catch (...) {
+    std::cerr << "node_id not specified in ET" << std::endl;
+  }
+  try {
+    node_name = data["workload_graph"][id]["Name"];
+  } catch (...) {
+    std::cerr << "node_name not specified in ET" << std::endl;
+  }
+  try {
+    node_type = data["workload_graph"][id]["NodeType"];
+  } catch (...) {
+    std::cerr << "node_type not specified in ET" << std::endl;
+  }
+  try {
+    is_cpu_op = data["workload_graph"][id]["is_cpu_op"];
+  } catch (...) {
+    std::cerr << "is_cpu_op not specified in ET" << std::endl;
+  }
+  try {
+    runtime = data["workload_graph"][id]["runtime"];
+  } catch (...) {
+  }
+  try {
+    data_deps =
+        data["workload_graph"][id]["data_deps"].get<std::vector<uint64_t>>();
+  } catch (...) {
+    std::cerr << "data deps not specified in ET" << std::endl;
+  }
+  /* 	COMM_SEND_NODE = 5,
```

Comment:
Let's use the autogenerated C++ files and variables in the code, rather than hardcoding values directly.